src/main/java/com/traveltime/sdk/dto/requests/proto/Transportation.java

```diff
@@ -1,35 +1,197 @@
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Getter;
 
 public interface Transportation {
-    String getValue();
-    Integer getCode();
+    TransportationType getType();
+
+    RequestsCommon.Transportation getProtoMessage();
+
+    class Modes {
+        public final static Transportation PUBLIC_TRANSPORT = PublicTransport.builder().build();
+        public final static Transportation DRIVING_AND_PUBLIC_TRANSPORT = DrivingAndPublicTransport.builder().build();
+        public final static Transportation WALKING_FERRY = new WalkingFerry();
+        public final static Transportation CYCLING_FERRY = new CyclingFerry();
+        public final static Transportation DRIVING_FERRY = new DrivingFerry();
+        public final static Transportation WALKING = new Walking();
+        public final static Transportation CYCLING = new Cycling();
+        public final static Transportation DRIVING = new Driving();
+    }
+
+    @Builder
+    @Getter
+    class PublicTransport implements Transportation {
+        public final TransportationType type = TransportationType.Types.PUBLIC_TRANSPORT;
+
+        @Builder.Default
+        public TransportationDetails.PublicTransportDetails details =
+            new TransportationDetails.PublicTransportDetails(1800);
+
+        @Override
+        public RequestsCommon.Transportation getProtoMessage() {
+            return RequestsCommon.Transportation
+                .newBuilder()
+                .setPublicTransport(
+                    RequestsCommon.PublicTransportDetails
+                        .newBuilder()
+                        .setWalkingTimeToStation(this.details.walkingTimeToStation)
+                )
+                .setTypeValue(this.type.getCode())
+                .build();
+        }
+    }
+
+    @Builder
+    @Getter
+    class DrivingAndPublicTransport implements Transportation {
+        public final TransportationType type = TransportationType.Types.DRIVING_AND_PUBLIC_TRANSPORT;
+
+        @Builder.Default
+        public final TransportationDetails.DrivingAndPublicTransportDetails details = new TransportationDetails.DrivingAndPublicTransportDetails(
+            1800, 1800, 300
+        );
+
+        @Override
+        public RequestsCommon.Transportation getProtoMessage() {
+            return RequestsCommon.Transportation
+                .newBuilder()
+                .setDrivingAndPublicTransport(
+                    RequestsCommon.DrivingAndPublicTransportDetails
+                        .newBuilder()
+                        .setWalkingTimeToStation(this.details.walkingTimeToStation)
+                        .setDrivingTimeToStation(this.details.drivingTimeToStation)
+                        .setParkingTime(this.details.parkingTime)
+                )
+                .setTypeValue(this.type.getCode())
+                .build();
+        }
+    }
+
+    @Getter
+    class WalkingFerry implements Transportation {
```

Comment:
For modes which don't have any special details you could just have a single
```

    @Value
    class SimpleTransportation implements Transportation {
        private final TransportationType type;

        @Override
        public RequestsCommon.Transportation getProtoMessage() {
            return RequestsCommon.Transportation.newBuilder().setTypeValue(this.type.getCode()).build();
        }
    }
```

and 

```
public final static Transportation DRIVING = new SimpleTransportation(TransportationType.Types.DRIVING);
```

Reply:
So the idea is that we'd want users to use:

```
    .transportation(Modes.DRIVING)
``` 
or 
```
  .transportation(Modes.PUBLIC_TRANSPORT.withTravelToStation(1800))
```
?

Reply:
Yes, I think that'd be ideal.

Reply:
I think we could take the "consenting adults" approach here, where it's possible for the user to write some funky code if necessary. For example, if we experiment with a "biking+public_transport" mode it'd be nice to be able to use it without having to release a new SDK version.

Reply:
I take it all back, the constructors should be protected.